maplibre/src/style/style.rs

```diff
@@ -16,117 +20,243 @@ use crate::style::{
 pub struct Style {
     pub version: u16,
     pub name: String,
-    pub metadata: HashMap<String, String>,
-    pub sources: HashMap<String, Source>,
-    pub layers: Vec<StyleLayer>,
-    pub center: Option<[f64; 2]>, // TODO: Use LatLon type here
+    pub metadata: Map<String, Value>,
+    pub center: Option<[f64; 2]>,
+    // pub center: Option<[f64; 2]>, // TODO: Use LatLon type here
     pub zoom: Option<f64>,
+    pub bearing: Option<f64>,
     pub pitch: Option<f64>,
+    pub light: Option<Light>,
+    pub terrain: Option<Map<String, Value>>,
+    pub sources: HashMap<String, Source>,
+    pub sprite: Option<String>,
+    pub glyphs: Option<String>,
+    pub transition: Option<Transition>,
+
+    // TODO is this in the style spec?
+    pub projection: Option<Map<String, Value>>,
+
+    pub layers: Vec<StyleLayer>,
+
+    // TODO this is metadata, and it's not in the spec
+    pub created: Option<String>,
+    pub modified: Option<String>,
+    pub id: Option<String>,
+    pub owner: Option<String>,
+    pub visibility: Option<String>,
+    pub protected: Option<bool>,
+    pub draft: Option<bool>,
```

Comment:
If this not in the spec, we might put this data into a Map<String, String>

Reply:
But maybe its better to have these here, in case they exist we can just parse them.